### ISSUE TITLE
Disable down scroll in the week area of the Agenda list and snap to each week

### ISSUE THREAD

**Issue by shamilovtim** [wix/react-native-calendars#951]
#### Issue Description

Downward scroll inside the Agenda causes user frustration and breaks the principle of least surprise. If the user is allowed to swipe down on the agenda in order to bring down the monthly view then the downward scroll interferes with user's swipe down touch event. Even if the swipe-down behavior were removed, the user still expects only to be able to scroll side to side within a weekly component, not down. Furthermore, the downward scrolling is not snapping to rows, which is causing user frustration.

![RPReplay_Final1570146103](https://user-images.githubusercontent.com/3495974/66213333-8bb23500-e68d-11e9-97fa-938988290d24.gif)

#### Immediate fixes:

1. Each weekly row should be snapped to with an animation. The scroll should not be allowed to stop in between a row.
2. Remove downscroll from the week view. The user should not be allowed to scroll inside the week view by swiping down inside the week component. 

#### Ideal functionality (distant future):

1.  Rather than up and down, the weekly component should move left to right 7 elements at a time.
2. Swiping down will bring down a month view that covers the screen.

**Comment by Grohden** [comment]
@shamilovtim not sure if it helps, but there's a `horizontal` prop on the `Agenda` that should let you control the scroll direction, you can probably pass `horizontal` on the jsx and it will make the **calendar** (not the weeks) scroll horizontally.

Not sure about `Agenda` but on `ExpandableCalendar` there's a issue with this prop ([I've fixed it in my fork](https://github.com/Grohden/react-native-calendars/commit/3cd41db8be1edb0af59624c3040b37d263b2e399#diff-e44193847bacdc4c318fbfd67dd883efL230-L254)). The issue is that, using horizontal there's no scroll down, so when you click on an arrow, it will change the week, but will not 'focus' on the week. My fix is when the horizontal prop is true, go to the next month instead of next week.

**Comment by shamilovtim** [comment]
No prop `horizontal` on Agenda according to the type definitions. What's important is that weeks don't go up and down. Weeks should go left and right.

<img width="883" alt="Screen Shot 2019-10-04 at 1 17 52 PM" src="https://user-images.githubusercontent.com/3495974/66227176-a515aa00-e6aa-11e9-8f88-4e7cf53e86d3.png">

Here is a component where weeks go left and right:
https://github.com/FaiChou/react-native-slideable-calendar-strip
![calendar-strip](https://user-images.githubusercontent.com/3495974/66227964-4cdfa780-e6ac-11e9-887d-623937450019.gif)

Here is another one:
https://github.com/BugiDev/react-native-calendar-strip
![parallel](https://user-images.githubusercontent.com/3495974/66228116-ac3db780-e6ac-11e9-9fcf-dcbc44d45dde.gif)

**Comment by shamilovtim** [comment]
As a prototype. here is Fantastical (proprietary, not RN) for iOS and Mac:
![Uploading 65351240-8c3ecc00-dbb5-11e9-9654-dbcf959b98d8.gif…]()

**Comment by shamilovtim** [comment]
@Grohden On your repo do you have most of the useful unmerged PRs from here merged? I was going to fork myself but if you're working on it then no need for me to double my efforts

**Comment by Grohden** [comment]
Typedefs can lie ( :'( ), but yeah, there's no prop on agenda :/ I've looked at the wrong source, sorry. 

About the merged PR's: nope, my intention was to merge it here and them help to maintain the project since I rewrote stuff to ts. I can try to merge them tho.. 

About the gif prototype, the `ExpandableCalendar` is almost the same... but more buggy than agenda.

**Comment by Grohden** [comment]
Btw I'm also rewriting (entirely) this lib using hooks + date-fns.I will publish it if I find that it could be useful for people

**Comment by shamilovtim** [comment]
There are a lot of solutions in the PRs for existing problems in the repo but so many of them haven't been merged or closed that I'm not sure what's up

**Comment by pi18** [comment]
Can we just prevent vertical scrolling when the calendar is collapsed?

**Comment by shamilovtim** [comment]
Yeah but unfortunately I don't foresee investing the time into a PR when there are 87 unmerged

**Comment by halk** [comment]
I noticed there is a typo here: https://github.com/wix/react-native-calendars/blob/master/src/agenda/index.js#L419

Renaming to `scrollEnabled` disabled the scrolling until the calendar list is opened

**Comment by ihsanktmr** [comment]
you can try my temporary solution https://github.com/wix/react-native-calendars/issues/998#issuecomment-570073538

**Comment by Inbal-Tish** [comment]
@shamilovtim If you need a week view calendar that scrolls left/right you can check out our ExpandableCalendar and the WeekCalendar.

**Comment by alindao-charles** [comment]
@Inbal-Tish Hi, I am having an issue with weekCalendar disabling the scroll event, is there a way to disable it? Thank you!

**Comment by chenei** [comment]
@shamilovtim @halk @Grohden @alindao-charles 
Are you still having problems?
Closing the issue for now.

**Comment by scouty-io** [comment]
> Btw I'm also rewriting (entirely) this lib using hooks + date-fns.I will publish it if I find that it could be useful for people

Would love to see what you did with your react-native-calendars

**Comment by Grohden** [comment]
@scouty-io the initial work is here if it helps:

https://github.com/Grohden/keepfy-react-native-calendars

But due to some time requirements most of the final work was not committed back to the lib and I'm not working at the company anymore so I can't sync final the sources.

But I guess you could try this one:
https://github.com/MarceloPrado/flash-calendar

Seems to be well maintained and finished
